storage/src/test/java/org/apache/kafka/tiered/storage/integration/EnableRemoteLogOnTopicTest.java

```diff
@@ -0,0 +1,89 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+
+public final class EnableRemoteLogOnTopicTest extends TieredStorageTestHarness {
+
+    @Override
+    public int brokerCount() {
+        return 2;
+    }
+
+    @Override
+    protected void writeTestSpecifications(TieredStorageTestBuilder builder) {
+        final Integer broker0 = 0;
+        final Integer broker1 = 1;
+        final String topicA = "topicA";
+        final Integer p0 = 0;
+        final Integer p1 = 1;
+        final Integer partitionCount = 2;
+        final Integer replicationFactor = 2;
+        final Integer maxBatchCountPerSegment = 1;
+        final boolean enableRemoteLogStorage = false;
+        final Map<Integer, List<Integer>> assignment = mkMap(
+                mkEntry(p0, Arrays.asList(broker0, broker1)),
+                mkEntry(p1, Arrays.asList(broker1, broker0))
+        );
+
+        builder
+                .createTopic(topicA, partitionCount, replicationFactor, maxBatchCountPerSegment, assignment,
+                        enableRemoteLogStorage)
+                // send records to partition 0
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 0L)
+                .produce(topicA, p0, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+                // send records to partition 1
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p1, 0L)
+                .produce(topicA, p1, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+                // enable remote log storage
+                .updateTopicConfig(topicA,
+                        Collections.singletonMap(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true"),
+                        Collections.emptyList())
+                // produce some more records to partition 0
+                // Note that the segment 0-2 gets offloaded for p0, but we cannot expect those events deterministically
+                // because the rlm-task-thread runs in background and this framework doesn't support it.
+                .expectSegmentToBeOffloaded(broker0, topicA, p0, 3, new KeyValueSpec("k3", "v3"))
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 4L)
+                .produce(topicA, p0, new KeyValueSpec("k3", "v3"), new KeyValueSpec("k4", "v4"))
+                // produce some more records to partition 1
+                // Note that the segment 0-2 gets offloaded for p1, but we cannot expect those events deterministically
+                // because the rlm-task-thread runs in background and this framework doesn't support it.
+                .expectSegmentToBeOffloaded(broker1, topicA, p1, 3, new KeyValueSpec("k3", "v3"))
```

Comment:
why do we need the 5th argument here is we are anyways producing it below in produce() call? The produce action must already be adding these to "offloadable" no?

Reply:
The 4th parameter is the base-offset of the segment and the 5th parameter is the contents of that segment. The test asserts that the segment contains only the expected messages in it. It is a `varargs` argument.